### PR TITLE
Create expo-turtle-build.md

### PR DESCRIPTION
**Pull request by godrei**
### Context

The default workflow for Expo projects triggers Builds on Expo build servers [using the EAS CLI](https://docs.expo.dev/build/introduction/).

This PR adds an alternative by building the IPA and AAB files on Bitrise [using Turtle CLI](https://docs.expo.dev/classic/turtle-cli/).

### Changes

- Introduce the `(Expo) Build using Turtle CLI` recipe.

### Investigation details

I followed Expo's [Building Standalone Apps on Your CI](https://docs.expo.dev/classic/turtle-cli/) guide.

### Decisions

As `fastlane` and `node` are preinstalled on each `linux-docker-android-*` and `osx-xcode-*` stack the recipe does not have a sophisticated mechanism for ensuring them (install if needed).

Both the recipe's and Expo project's dependencies are installed via `npm` for the sake of simplicity (instead of `yarn`).

### CODE REVIEW

File: recipes/expo-turtle-build.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# (Expo) Build using Turtle CLI
+
+## Description
+
+Publish an app to Expo's servers and build an iOS App Store .ipa and Android .aab files from your Expo project using [Turtle CLI](https://docs.expo.dev/classic/turtle-cli/).
```
Comment (shams-ahmed) [inline recipes/expo-turtle-build.md]: NIT: `.ipa` and .aab highlight?

**Review by shams-ahmed — APPROVED** [review]

File: recipes/expo-turtle-build.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    - file-downloader@1:
+        title: Download iOS Provisioning Profile
+        inputs:
+        - destination: $PROFILE_PATH
+        - source: $BITRISE_PROVISION_URL
+    - npm@1:
+        title: Install project dependencies
+        inputs:
+        - command: install
+    - set-java-version@1:
```
Comment (ofalvai) [inline recipes/expo-turtle-build.md]: Is this really needed in 2022? 😣

**Review by ofalvai — APPROVED** [review]

File: recipes/expo-turtle-build.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
+# (Expo) Build using Turtle CLI
```
Comment (chreno) [inline recipes/expo-turtle-build.md]:
```suggestion
# (React Native) Expo: Build using Turtle CLI
```

File: recipes/expo-turtle-build.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    - KEYSTORE_PATH: /tmp/keystore.jks
+    - KEYSTORE_ALIAS: $BITRISEIO_ANDROID_KEYSTORE_ALIAS
+    - EXPO_ANDROID_KEYSTORE_PASSWORD: $BITRISEIO_ANDROID_KEYSTORE_PASSWORD
+    - EXPO_ANDROID_KEY_PASSWORD: $BITRISEIO_ANDROID_KEYSTORE_PRIVATE_KEY_PASSWORD
+    - PROFILE_PATH: /tmp/profile.mobileprovision
+    - CERTIFICATE_PATH: /tmp/certificate.p12
+    - EXPO_IOS_DIST_P12_PASSWORD: $BITRISE_CERTIFICATE_PASSPHRASE
+    - IOS_DEVELOPMENT_TEAM: $IOS_DEVELOPMENT_TEAM
+    - EXPO_USERNAME: $EXPO_USERNAME
+    - EXPO_PASSWORD: $EXPO_PASSWORD
```
Comment (chreno) [inline recipes/expo-turtle-build.md]: Do we need to add the secrets as well to the envs?

File: recipes/expo-turtle-build.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+2. Generate an Android Keystore by following the [Android code signing with Android Studio](https://devcenter.bitrise.io/en/code-signing/android-code-signing/android-code-signing-with-android-studio.html) guide.
+3. Make sure you can [Publish your Expo project](https://docs.expo.dev/classic/turtle-cli/#publish-your-project) locally.
+
+## Instructions
+
+1. Upload the project's iOS Distribution Certificate and App Store Provisioning Profile on the Bitrise project's Workflow Editor / Code signing tab.
+2. Upload the project's Android Keystore on the Bitrise project's Workflow Editor / Code signing tab.
+3. Create a Secret (`IOS_DEVELOPMENT_TEAM`) with the ID of the iOS Development Team, issued the project's Certificate and Provisioning Profile.
+4. Store the Expo account, used for publishing the Expo app and fetching the app manifest, in `EXPO_USERNAME` and `EXPO_PASSWORD` secrets.
+5. Copy-paste `envs` to your workflow.
```
Comment (chreno) [inline recipes/expo-turtle-build.md]:
```suggestion
5. Copy-paste `envs` from `bitrise.yml` below to your Workflow.
```

File: recipes/expo-turtle-build.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+6. Copy-paste `steps` to your workflow.
```
Comment (chreno) [inline recipes/expo-turtle-build.md]:
```suggestion
6. Copy-paste `steps` from `bitrise.yml` below to your Workflow.
```

File: recipes/expo-turtle-build.md (hunk 3 above)
Comment (chreno) [reply recipes/expo-turtle-build.md]: I also propose to prefix the file name with `rn-` for the sake of consistency.

File: README.md [hunk 7]
```diff
@@ -50,6 +50,10 @@ You can also simply copy-paste the snippet to your `bitrise.yml` file directly.
 * [(React Native) Run Tests](recipes/rn-tests.md)
 * [(Flutter) Run Tests](recipes/flutter-test.md)
 
+### Building
+
+* [(Expo) Build using Turtle CLI](recipes/expo-turtle-build.md)
```
Comment (chreno) [inline README.md]:
```suggestion
* [(React Native) Expo: Build using Turtle CLI](recipes/expo-turtle-build.md)
```

File: recipes/expo-turtle-build.md (hunk 2 above)
Comment (godrei) [reply recipes/expo-turtle-build.md]: unfortunately yes: https://docs.expo.dev/classic/turtle-cli/#for-android-builds

File: recipes/expo-turtle-build.md (hunk 4 above)
Comment (godrei) [reply recipes/expo-turtle-build.md]: Not really, this way I just wanted to highlight all the envs required to run the recipe.
In the case of the steps' sample workflows, we did a similar thing, we also added a comment above these envs: `Define these in your secrets`.

I'm not sure if it helps, so I'm happy to remove the secrets if you think.

Comment (chreno) [reply recipes/expo-turtle-build.md]: I see, I think it doesn't hurt to have them here. I like the comment idea. Should we add the comment here as well?

Comment (godrei) [reply recipes/expo-turtle-build.md]: kept them and added a comment (`Define these in your secrets`)

File: recipes/expo-turtle-build.md (hunk 1 above)
Comment (godrei) [reply recipes/expo-turtle-build.md]: Added code format in: https://github.com/bitrise-io/workflow-recipes/pull/1/commits/aa8469ff2ff570219625c45ff41c93d815e1544b

**Review by chreno — APPROVED** [review]

**Pull request merged**